touchforms/formplayer/static/formplayer/script/webformsession.js

```diff
@@ -142,31 +142,35 @@ WebFormSession.prototype.serverRequest = function (requestParams, callback, bloc
     this.numPendingRequests++;
     this.onLoading();
 
-    if(self.formplayerEnabled){
+    if (self.formplayerEnabled){
         $.ajax({
-                type: 'POST',
-                url: url + "/" + requestParams.action,
-                data: JSON.stringify(requestParams),
-                contentType: "application/json",
-                dataType: "json",
-                crossDomain: {crossDomain: true},
-                xhrFields: {withCredentials: true},
-            })
-            .success(function (resp) {
+            type: 'POST',
+            url: url + "/" + requestParams.action,
+            data: JSON.stringify(requestParams),
+            contentType: "application/json",
+            dataType: "json",
+            crossDomain: {crossDomain: true},
+            xhrFields: {withCredentials: true},
+            success: function(resp) {
                 self.handleSuccess(resp, callback);
-            })
-            .fail(function (resp, textStatus) {
+            },
+            error: function(resp, textStatus) {
                 self.handleFailure(resp, textStatus);
-            });
-    } else{
+            },
+        });
+    } else {
         $.ajax({
-                type: 'POST',
-                url: url,
-                data: JSON.stringify(requestParams),
-                dataType: "text"  // we don't use JSON because of a weird bug: http://manage.dimagi.com/default.asp?190983
-            })
-            .success(function(resp) { self.handleSuccess(JSON.parse(resp), callback); })
-            .fail(function (resp, textStatus) { self.handleFailure(JSON.parse(resp), textStatus); });
+            type: 'POST',
+            url: url,
+            data: JSON.stringify(requestParams),
+            dataType: "text",  // we don't use JSON because of a weird bug: http://manage.dimagi.com/default.asp?190983
+            success: function(resp) {
```

Comment:
It's just the success/error/complete functions being removed (https://jquery.com/upgrade-guide/3.0/#breaking-change-special-case-deferred-methods-removed-from-jquery-ajax). I could have left the `fail` call here alone, but my understanding is that moving that code into an error callback instead will behave the same way and is the more common way that HQ code is written.

3.0 does make a bunch of changes to Deferred objects (https://jquery.com/upgrade-guide/3.0/#deferred). They're some of the changes I'm most worried about in upgrading...we don't do a lot of promise-based code, but it's difficult to look at our code and know if it'll be affected.